eth/vm/forks/london/headers.py

```diff
@@ -0,0 +1,144 @@
+    compute_berlin_difficulty,
+    configure_header,
+)
+
+from .blocks import LondonBlockHeader
+from .constants import (
+    BASE_FEE_MAX_CHANGE_DENOMINATOR,
+    INITIAL_BASE_FEE,
+)
+
+
+def calculate_expected_base_fee_per_gas(parent_header: BlockHeaderAPI) -> int:
+    if parent_header is None:
+        # Parent is empty when making the genesis header
+        parent_base_fee_per_gas = None
+    else:
+        parent_base_fee_per_gas = parent_header.base_fee_per_gas
+
+    if parent_base_fee_per_gas is None:
+        # Parent is a non-London header or this is a genesis header
+        return INITIAL_BASE_FEE
+
+    parent_gas_target = parent_header.gas_limit
+    parent_gas_used = parent_header.gas_used
+
+    if parent_gas_used == parent_gas_target:
+        return parent_base_fee_per_gas
+
+    elif parent_gas_used > parent_gas_target:
+        gas_used_delta = parent_gas_used - parent_base_fee_per_gas
+        base_fee_per_gas_delta = max(
+            (
+                parent_base_fee_per_gas * gas_used_delta
+                // parent_gas_target // BASE_FEE_MAX_CHANGE_DENOMINATOR
+            ),
+            1
+        )
+        return parent_base_fee_per_gas + base_fee_per_gas_delta
+
+    else:
+        gas_used_delta = parent_gas_target - parent_gas_used
+        base_fee_per_gas_delta = parent_base_fee_per_gas * gas_used_delta \
+            // parent_gas_target // BASE_FEE_MAX_CHANGE_DENOMINATOR
+        return max(parent_base_fee_per_gas - base_fee_per_gas_delta, 0)
+
+
+@curry
+def create_header_from_parent(difficulty_fn: Callable[[BlockHeaderAPI, int], int],
+                              parent_header: Optional[BlockHeaderAPI],
+                              **header_params: Any) -> BlockHeaderAPI:
+
+    # frontier
+    if 'gas_limit' not in header_params:
+        header_params['gas_limit'] = compute_gas_limit(
+            parent_header,
+            genesis_gas_limit=GENESIS_GAS_LIMIT,
+        )
+
+    # byzantium
+    if 'timestamp' not in header_params:
+        header_params['timestamp'] = new_timestamp_from_parent(parent_header)
+
+    if 'difficulty' not in header_params:
+        if parent_header is None:
+            raise ValueError(
+                "Must set difficulty when creating a new genesis header (no parent)."
+                " Consider 1 for easy mining or eth.constants.GENESIS_DIFFICULTY for consistency."
+            )
+        else:
+            header_params['difficulty'] = difficulty_fn(
+                parent_header,
+                header_params['timestamp'],
+            )
+
+    all_fields = fill_header_params_from_parent(parent_header, **header_params)
+
+    # must add the new field *after* filling, because the general fill function doesn't recognize it
+    if 'base_fee_per_gas' in header_params:
+        all_fields['base_fee_per_gas'] = header_params['base_fee_per_gas']
+    else:
+        all_fields['base_fee_per_gas'] = calculate_expected_base_fee_per_gas(parent_header)
```

Comment:
Either:
- stop allowing this to be passed in (a valid header must have an exactly calculated base fee)
- allow it to be passed in, but raise a validation error if it doesn't match the calculated value